### PR TITLE
Add pattern validation to operatorspec managementstate

### PR DESCRIPTION
**Pull request by damemi**

### CODE REVIEW

**Comment by damemi** [comment]
cc @sttts

**Comment by sttts** [comment]
LGTM if this passed the smoke test.

**Comment by sttts** [comment]
Worked in a cluster according to @damemi

File: operator/v1/types.go [hunk 1]
```diff
@@ -44,6 +44,7 @@ var (
 type OperatorSpec struct {
 	// managementState indicates whether and how the operator should manage the component
 	// +optional
+	// +kubebuilder:validation:Pattern=^Managed|Unmanaged|Force|Removed$
```
Comment (sttts) [inline operator/v1/types.go]: remove optional

**Pull request merged**